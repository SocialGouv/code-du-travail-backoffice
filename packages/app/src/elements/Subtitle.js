import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";

const StyledSubtitle = styled.h2`
  color: black;
  font-size: 1.2rem;
  font-weight: 600;
  margin: ${props => (props.isFirst ? "0 0 0.5rem" : "1.5rem 0 0.5rem")};
  user-select: none;
`;

const Subtitle = ({ isFirst = false, ...props }) => <StyledSubtitle isFirst={isFirst} {...props} />;

Subtitle.propTypes = {
  isFirst: PropTypes.bool,
};

export default Subtitle;
