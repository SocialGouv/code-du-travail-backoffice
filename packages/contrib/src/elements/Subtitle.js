import styled from "styled-components";

export default styled.h2`
  color: black;
  font-size: 1.1rem;
  font-weight: 600;
  margin: ${props => (props.isFirst ? "0 0 0.5rem" : "1.5rem 0 0.5rem")};
  user-select: none;
`;
