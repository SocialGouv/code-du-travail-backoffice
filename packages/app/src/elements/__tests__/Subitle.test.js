import React from "react";

import Subtitle from "../Subtitle";

describe("elements/<Subtitle />", () => {
  const TEXT = "A Subtitle";

  it(`should pass`, () => {
    const $subtitle = testRender(<Subtitle>{TEXT}</Subtitle>);

    expect($subtitle).toMatchSnapshot();
    expect($subtitle).toHaveStyleRule("margin", "1.5rem 0 0.5rem");
  });

  it(`should pass with {isFirst}`, () => {
    const $subtitle = testRender(<Subtitle isFirst>{TEXT}</Subtitle>);

    expect($subtitle).toHaveStyleRule("margin", "0 0 0.5rem");
  });
});
