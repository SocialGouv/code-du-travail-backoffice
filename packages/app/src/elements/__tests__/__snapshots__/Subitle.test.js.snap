// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/<Subtitle /> should pass 1`] = `
.emotion-0 {
  color: black;
  font-size: 1.2rem;
  font-weight: 600;
  margin: 1.5rem 0 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<h2
  className="emotion-0 emotion-1"
>
  A Subtitle
</h2>
`;
